packaging/setup/plugins/ovirt-engine-setup/vmconsole_proxy_helper/pki.py

```diff
@@ -22,17 +22,20 @@
 
 from ovirt_engine_setup import constants as osetupcons
 from ovirt_engine_setup.engine import constants as oenginecons
+from ovirt_engine_setup.engine_common import pki_utils
 from ovirt_engine_setup.vmconsole_proxy_helper import constants as ovmpcons
 
 
 def _(m):
     return gettext.dgettext(message=m, domain='ovirt-engine-setup')
 
 
-def _refresh_needed(cert_path):
+def _refresh_needed(cert_path, check_cert=True):
     ca_cert_path = oenginecons.FileLocations.OVIRT_ENGINE_PKI_ENGINE_CA_CERT
     return (not os.path.exists(cert_path) or
-            os.stat(ca_cert_path).st_mtime > os.stat(cert_path).st_mtime)
+            os.stat(ca_cert_path).st_mtime > os.stat(cert_path).st_mtime or
```

Comment:
IIRC this check might be recommended by you.  The idea is that if the CA certificate is touched then it may be changed and we must renew in such a case to get the vmconsole certificate signed by the new CA certificate.  If a user touches the CA certificate just for fun or so then we renew unnecessarily but yes, this is harmless.  The time check is in theory not absolutely reliable but it's simple and good enough I think (or do we already have a better way?).